tests/test_activate.py

```diff
@@ -1,652 +1,582 @@
+
+else:
+    # Only run these tests for commands that are installed.
+    shells = ['bash', 'zsh']
+    ps_var = "PS1"
+    var_format = "${var}"
+    binpath = "/bin/"  # mind the trailing slash.
+    source_setup = "source"
+    nul = '2>/dev/null'
+    set_var = 'export '
+    shell_suffix = ""
+    printps1 = '{echo} {var}'.format(echo=echo, var=var_format.format(var=ps_var))
+
+for shell in shells[:]:
+    try:
+        stdout, stderr = run_in('echo', shell)
+    except OSError:
+        pass
+    else:
+        if not stderr:
+            shells.append(shell)
+
+raw_ps = escape_curly(os.getenv(ps_var, ""))
+def print_ps1(env_dirs, number):
+    return u" ".join([u"({})".format(os.path.split(env_dirs[number])[-1]), raw_ps]).strip()
+
+CONDA_ENTRY_POINT = """\
 #!{syspath}/python
 import sys
 from conda.cli import main
 
 sys.exit(main())
 """
+printpath = '{echo} {var}'.format(echo=echo, var=var_format.format(var="PATH"))
+if sys.platform == "win32":
+    printdefaultenv = '{echo}.{var}'.format(echo=echo, var=var_format.format(var="CONDA_ACTIVE_ENV"))
+else:
+    printdefaultenv = '{echo} {var}'.format(echo=echo, var=var_format.format(var="CONDA_ACTIVE_ENV"))
 
 command_setup = """\
-export PATH="{ROOTPATH}"
-export PS1='$'
-export PYTHONPATH="{PYTHONPATH}"
-export CONDARC=' '
-cd {here}
-""".format(here=dirname(__file__), ROOTPATH=ROOTPATH, PYTHONPATH=PYTHONPATH)
-
-command_setup = command_setup + """
-mkdir -p {envs}/test1/bin
-mkdir -p {envs}/test2/bin
-"""
-
+set {ps_var}={raw_ps}
+set PYTHONPATH={PYTHONPATH}
+set CONDARC=
+""".format(here=dirname(__file__), PYTHONPATH=PYTHONPATH,
+           ps_var=ps_var, raw_ps=raw_ps)
+
+if platform.startswith("win"):
+    command_setup = "@echo off\n" + command_setup
+
+
+_format_vars = {
+    'nul': nul,
+    'printpath': printpath,
+    'printdefaultenv': printdefaultenv,
+    'printps1': printps1,
+    'set_var': set_var,
+    'source': source_setup,
+    'binpath': binpath,
+    'syspath': sys.prefix,
+    'shell_suffix': shell_suffix
+}
 
 @pytest.mark.slow
 def test_activate_test1():
     for shell in shells:
         with TemporaryDirectory(prefix='envs', dir=dirname(__file__)) as envs:
-            activate, deactivate, conda = _write_entry_points(envs)
             commands = (command_setup + """
-            source {activate} {envs}/test1
-            printf $PATH
-            """).format(envs=envs, activate=activate)
+            {source} {syspath}{binpath}activate{shell_suffix} "{env_dirs[0]}"
+            {printpath}
+            """).format(envs=envs, env_dirs=gen_test_env_paths(envs), **_format_vars)
 
             stdout, stderr = run_in(commands, shell)
-            assert stdout == envs + "/test1/bin:" + PATH
-            assert stderr == 'discarding {syspath} from PATH\nprepending {envs}/test1/bin to PATH\n'.format(envs=envs, syspath=syspath)
+            assert_equals(stderr, u'prepending {envpaths} to PATH'\
```

Comment:
Hm... I see. I don't have any objections against keeping the PATH clean. However there is one important thing:

**There must be a (command line) way to put conda/activate on the PATH.**

In Visual Studio, there is always a `vcars*.bat` or `setenv.cmd` script on the PATH which allows you to quickly add the development environment to the shell you are currently working with. Although there is start menu command to open such a shell in a new terminal, it is *essential* to be able to do that from a script (e.g. Makefile).

For Anaconda, it could work like this:
1. Skip adding `Anaconda\Scripts` and `Anaconda` to the PATH when installing Anaconda.
2. Provide a `anaconda.bat` script in `Anaconda\somepath` which adds `Anaconda\Scripts` and `Anaconda` to the PATH.
3. Put `Anaconda\somepath` on the system PATH

This allows to easily provide a start menu entry (just call `cmd /k anaconda.bat`) but also to integrate Anaconda in custom batch scripts. In my company we have a lot of scripts like the following Jupyter notebook starter (simplified code):
```batch
if not exist "%~dp0venv" (
    conda create -p %~dp0venv --file=requirements.txt jupyter
)
call activate %~dp0venv
jupyter notebook
```
Simply by distributing this batch script together with a requirements file, we can provide the user a single entry point to installing and running Jupyter notebook. If you decide to drop Anaconda from the system path, we have to adapt this code as conda is not on the PATH. The easiest solution would be to add a `call anaconda.bat` at the beginning of the script.